ui/src/App.tsx

```diff
@@ -1,15 +1,52 @@
 import React from 'react'
 import './App.css'
 import { getDashboardCounters } from './lib/api'
+import routie from './lib/routie'
 import setIntervalDebounced from './lib/setIntervalDebounced'
+import { DashboardCounters } from './models/dashboardCounters'
 
-interface IState { }
+interface IState { 
+  renderMethod: () => JSX.Element
+  dashboardCounters: DashboardCounters
+}
 
 export default class App extends React.Component<{}, IState> {
+
+  state:IState = {
+    renderMethod: () => <div>Loading...</div>,
+    dashboardCounters: {
+      simpleGrainStats: [],
+      totalActivationCount: 0,
+      totalActivationCountHistory: [],
+      totalActiveHostCount: 0,
+      totalActiveHostCountHistory: [],
+      hosts: []
+    }
+  }
+
   cancel?: () => void
 
   componentDidMount() {
     this.cancel = setIntervalDebounced(this.loadDataOnSchedule, 1000)
+
+    routie('', () => {
```

Comment:
I would use react-router like everybody else ;)

Reply:
Yes, I probably should! I despise using it though, almost as much as redux.

Reply:
Harsh words...why?